SIPS/sip-56.md

```diff
@@ -0,0 +1,94 @@
+discussions-to: <https://discordapp.com/invite/AEdUHzt>
+
+created: 2020-05-01
+---
+
+<!--You can leave these HTML comments in your merged SIP and delete the visible duplicate text guides, they will not appear and may be helpful to refer to if you edit it again. This is the suggested template for new SIPs. Note that an SIP number will be assigned by an editor. When opening a pull request to submit your SIP, please use an abbreviated title in the filename, `sip-draft_title_abbrev.md`. The title should be 44 characters or less.-->
+
+## Simple Summary
+
+<!--"If you can't explain it simply, you don't understand it well enough." Provide a simplified and layman-accessible explanation of the SIP.-->
+
+Ability for the protocol DAO to configure different exchange fee rates for each `Synth`.
+
+## Abstract
+
+<!--A short (~200 word) description of the technical issue being addressed.-->
+
+Currently there is one universal exchange rate for exchanging synths which is set onchain in the `FeePool`.
+
+Upgrading the `FeePool` to allow an array of Synths exchange rates to be set allowing unique pricing per asset or class.
+
+There could be multiple pricing options TBD.
+
+1. Charge the fee rate of the synth being exchanged into. e.g. If sETH is 1% and sUSD was 0% then it would be free to exchange from sETH to sUSD.
+2. Charge the sum of the pair. e.g. if sETH is .5% and sUSD is .1% the total exchange fee would be 0.6%
+
+## Motivation
+
+<!--The motivation is critical for SIPs that want to change Synthetix. It should clearly explain why the existing protocol specification is inadequate to address the problem that the SIP solves. SIP submissions without sufficient motivation may be rejected outright.-->
+
+With a universal exchange fee rate the protocol is limited in it's ability to price markets fairly or apply incentive mechanisms. To deter front runners setting a high exchange rate was used to reduce profiitability affecting all Synth markets and discouraging good actors from trading.
+
+Unique pricing per synth would allow potential outcomes such as;
+
+a) Different price categories such as fiat, crypto, inverse, indices, commodities or stocks.
+
+b) Used as a configurable protocol mechanism via SCCP to increase rates to deter new trades exchanging into a synth where there is an oversupply or potential debt risk to the system.
+
+c) Configure a synth's exchange rate such as sUSD as low as possible to encourage traders to move into it or for SNX stakers to return to sUSD at no cost to be able to burn it to maintain the systems target c-ratio.
+
+## Specification
+
+<!--The technical specification should describe the syntax and semantics of any new feature.-->
+
+Storage of the rates should use the `uint mappings` in the `FeePoolEternalStorage` contract so `FeePool` logic can be upgraded maintaining the exchange fee rate state.
+
+New functions required in the upgradable `FeePool` contract.
+
+### setExchangeFeeRateForSynths
```

Comment:
This is really good but what about how the fees are calculated?

Reply:
Added the definition for exchangeFeeForTrade and to deprecate existing fee view function stub`FeePool.exchangeFeeIncurred`